src/controllers/__tests__/authenticationController.test.ts

```diff
@@ -531,6 +607,257 @@ describe("AuthenticationController#acs", () => {
       expectedNewProfile
     );
   });
+
+  it(`redirects to the correct url using the lollipop features`, async () => {
+    const res = mockRes();
+
+    mockGetLollipop.mockResolvedValueOnce(E.right(O.some(anAssertionRef)));
+    mockDelLollipop.mockResolvedValueOnce(E.right(true));
+    mockActivateLolliPoPKey.mockImplementationOnce((newAssertionRef, __, ___) =>
+      TE.of({
+        ...anActivatedPubKey,
+        assertion_ref: newAssertionRef
+      } as ActivatedPubKey)
+    );
+    mockSetLollipop.mockImplementationOnce((_, __, ___) =>
+      Promise.resolve(E.right(true))
+    );
+    mockSet.mockReturnValue(Promise.resolve(E.right(true)));
+    mockIsBlockedUser.mockReturnValue(Promise.resolve(E.right(false)));
+    mockGetNewToken
+      .mockReturnValueOnce(mockSessionToken)
+      .mockReturnValueOnce(mockWalletToken)
+      .mockReturnValueOnce(mockMyPortalToken)
+      .mockReturnValueOnce(mockBPDToken)
+      .mockReturnValueOnce(mockZendeskToken)
+      .mockReturnValueOnce(mockFIMSToken)
+      .mockReturnValueOnce(aSessionTrackingId);
+
+    mockGetProfile.mockReturnValue(
+      ResponseSuccessJson(mockedInitializedProfile)
+    );
+    const response = await lollipopActivatedController.acs(validUserPayload);
+    response.apply(res);
+
+    expect(lollipopActivatedController).toBeTruthy();
+    expect(res.redirect).toHaveBeenCalledWith(
+      301,
+      "/profile.html?token=" + mockSessionToken
+    );
+
+    expect(mockDelLollipop).toBeCalledWith(aFiscalCode);
+    expect(mockActivateLolliPoPKey).toBeCalledWith(
+      anotherAssertionRef,
+      aFiscalCode,
+      aLollipopAssertion,
+      expect.any(Function)
+    );
+    expect(mockSetLollipop).toBeCalledWith(
+      expect.objectContaining({
+        fiscal_code: aFiscalCode
+      }),
+      anotherAssertionRef
+    );
+    expect(mockRevokePreviousAssertionRef).toHaveBeenCalledWith(anAssertionRef);
```

Comment:
this should be the first call to be checked

Reply:
fixed in 1edce3c